sound/soc/sof/core.c

```diff
@@ -119,6 +119,19 @@ struct snd_sof_widget *snd_sof_find_swidget_sname(struct snd_sof_dev *sdev,
 	return NULL;
 }
 
+bool snd_sof_pcm_d0i3_suspend(struct snd_sof_dev *sdev)
```

Comment:
@keyonjie fix commit message:
"Add helper to check if the DSP should be put in D0i3. This function returns true if there a stream has ignored
the SUSPEND trigger to keep the pipeline running in the DSP"

Reply:
and why call it pcm_D0i3_suspend? I think snd_sof_dsp_D0i3_suspend() makes better sense no?

Reply:
> and why call it pcm_D0i3_suspend? I think snd_sof_dsp_D0i3_suspend() makes better sense no?

The intention was to emphasize that it is from the view of the stream/pcm point as what we were doing inside is iterating the stream list only.

Reply:
> @ranj063 @keyonjie is there a goto document to understand PM states. I must admit I have little experience with this.

Oh yes @dbaluta it is in my last commit, we draws a state machine diagram there, see?

Reply:
> The intention was to emphasize that it is from the view of the stream/pcm point as what we were doing inside is iterating the stream list only.

Right but what you're setting is the state for the DSP and not the pcm. So I dont think pcm in the name is relevant

Reply:
OK, let me change it.